server/routes/questionsRoute.test.ts

```diff
@@ -0,0 +1,216 @@
+import type { Express } from 'express'
+import request, { type Agent } from 'supertest'
+
+import { appWithAllRoutes } from './testutils/appSetup'
+import { IncidentReportingApi, ReportBasic, ReportWithDetails } from '../data/incidentReportingApi'
+import { convertReportWithDetailsDates } from '../data/incidentReportingApiUtils'
+import { mockReport } from '../data/testData/incidentReporting'
+
+jest.mock('../data/incidentReportingApi')
+
+let app: Express
+let incidentReportingApi: jest.Mocked<IncidentReportingApi>
+
+beforeEach(() => {
+  app = appWithAllRoutes({})
+  incidentReportingApi = IncidentReportingApi.prototype as jest.Mocked<IncidentReportingApi>
+})
+
+afterEach(() => {
+  jest.resetAllMocks()
+})
+
+describe('Displaying responses', () => {
+  const incidentDateAndTime = new Date('2024-10-21T16:32:00+01:00')
+  const reportWithDetails: ReportWithDetails = convertReportWithDetailsDates(
+    mockReport({
+      type: 'FINDS',
+      reportReference: '6544',
+      reportDateAndTime: incidentDateAndTime,
+      withDetails: true,
+    }),
+  )
+
+  const reportQuestionsUrl = `/reports/${reportWithDetails.id}/questions`
+
+  let agent: Agent
+
+  beforeEach(() => {
+    agent = request.agent(app)
+    incidentReportingApi.getReportById.mockResolvedValue(reportWithDetails as ReportBasic)
+    incidentReportingApi.getReportWithDetailsById.mockResolvedValue(reportWithDetails)
+  })
+
+  it('form is prefilled with report answers, including date', () => {
+    reportWithDetails.type = 'DEATH_OTHER'
```

Comment:
danger! shared `const reportWithDetails` being modified! when we add more tests to this file, we might not recall that type and questions must be written in every test case.

safer to either copy it, or make the `reportWithDetails` be a let variable populated with `beforeEach`.

Reply:
I'll have a look but tbh I'm setting the state just within the given test so it should be fine.

Reply:
if u don't want to create the report in a `beforeEach`, can u leave a comment next to the `reportWithDetails` declaration noting that _every_ test must set the type and questions?